### PR TITLE
Add binding for SHA256 ECDH

### PR DESCRIPTION
**Pull request by matsjj**

### CODE REVIEW

**Comment by matsjj** [comment]
@paulo-bc do I need to increase some version for this to be included into iOS? Obviously we don't want random changes like this to pop up downstream without us explicitly opting into it.

**Comment by Sjors** [comment]
Lgtm, but I only looked very superficially. This PR strictly adds the `ecdh` method, so nobody would accidentally use it.

File: LibWally/Address.swift [hunk 1]
```diff
@@ -254,6 +254,35 @@ public struct Key {
             return PubKey(Data(bytes: bytes_out, count: Int(EC_PUBLIC_KEY_LEN)), network, compressed: true)!
         }
     }
+    
+    public func ecdh(pubKey: PubKey) -> Data? {
```
Comment (jackpooleybc) [inline LibWally/Address.swift:258]: It might be nice to either make this `throw` or return a `Result` here which could provide specific error.

**Review by jackpooleybc — APPROVED** [review]
LGTM

File: LibWally/Address.swift (hunk 1 above)
Comment (matsjj) [reply LibWally/Address.swift:258]: Hm right, but this seems to be consistent with the rest of the library.

**Pull request merged**